# **The World Bank's International Debt Data**

<span style="color: rgba(0, 0, 0, 0.87); font-family: -apple-system, system-ui, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; background-color: rgb(255, 255, 255);">It's not that we humans only take debts to manage our necessities. A country may also take debt to manage its economy. For example, infrastructure spending is one costly ingredient required for a country's citizens to lead comfortable lives.&nbsp;</span> [The World Bank](https://www.worldbank.org/en/home) <span style="color: rgba(0, 0, 0, 0.87); font-family: -apple-system, system-ui, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; background-color: rgb(255, 255, 255);">&nbsp;is the organization that provides debt to countries.</span>

In this notebook, we are going to analyze international debt data collected by The World Bank. The dataset contains information about the amount of debt (in USD) owed by developing countries across several categories. We are going to find the answers to questions like:

- What is the total amount of debt that is owed by the countries listed in the dataset?
- Which country owns the maximum amount of debt and what does that amount look like?
- What is the average amount of debt owed by countries across different debt indicators?

This is part of [datacamp](/Users/rafsanalmamun/Downloads/Data-Analysis_Data-Science_Projects/SQL%20Projects/WB%20International%20Debt%20Statistics/dataset/www.datacamp.com)'s "[Analyze International Debt Statistics](https://www.datacamp.com/projects/754)" project.

### **\*\* Use the IDS Database \*\***

In [2]:
USE IDS;
GO

Commands completed successfully.

Total execution time: 00:00:00.009

## 1\. Overview of the 'international\_debt' Table

<span style="color: rgba(0, 0, 0, 0.87); font-family: -apple-system, system-ui, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; background-color: rgb(255, 255, 255);">Let's first&nbsp;</span> `SELECT` _all_ <span style="color: rgba(0, 0, 0, 0.87); font-family: -apple-system, system-ui, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; background-color: rgb(255, 255, 255);">of the columns from the&nbsp;</span> `international_debt` <span style="color: rgba(0, 0, 0, 0.87); font-family: -apple-system, system-ui, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; background-color: rgb(255, 255, 255);">table. Also, we'll limit the output to the first ten rows to keep the output clean.</span>

In [4]:
SELECT TOP(10) * FROM international_debt;

(10 rows affected)

Total execution time: 00:00:00.026

country_name country_code indicator_name indicator_code debt Afghanistan AFG PPG, bilateral (AMT, current US$) DT.AMT.BLAT.CD 61739336.9 Afghanistan AFG Principal repayments on external debt, long-term (AMT, current US$) DT.AMT.DLXF.CD 100847181.9 Afghanistan AFG PPG, multilateral (AMT, current US$) DT.AMT.MLAT.CD 39107845 Afghanistan AFG PPG, official creditors (AMT, current US$) DT.AMT.OFFT.CD 100847181.9 Afghanistan AFG PPG, bilateral (DIS, current US$) DT.DIS.BLAT.CD 49114729.4 Afghanistan AFG Disbursements on external debt, long-term (DIS, current US$) DT.DIS.DLXF.CD 72894453.7 Afghanistan AFG PPG, multilateral (DIS, current US$) DT.DIS.MLAT.CD 23779724.3 Afghanistan AFG PPG, official creditors (DIS, current US$) DT.DIS.OFFT.CD 72894453.7 Afghanistan AFG PPG, bilateral (INT, current US$) DT.INT.BLAT.CD 39903620.1 Afghanistan AFG Interest payments on external debt, long-term (INT, current US$) DT.INT.DLXF.CD 53239440.1

## 2\. Finding the Number of Distinct Countries

From the first ten rows, we can see the amount of debt owed by _Afghanistan_ in the different debt indicators. But we do not know the number of different countries we have on the table. There are repetitions in the country names because a country is most likely to have debt in more than one debt indicator.

Without a count of unique countries, we will not be able to perform our statistical analyses holistically. In this section, we are going to extract the number of unique countries present in the table.

In [5]:
SELECT COUNT(DISTINCT country_name) AS total_num_countries 
FROM international_debt;

(1 row affected)

Total execution time: 00:00:00.052

total_num_countries 124

## 3\. Finding Out the Distinct Debt Indicators

We can see there are a total of 124 countries present on the table. As we saw in the first section, there is a column called `indicator_name` that briefly specifies the purpose of taking the debt. Just beside that column, there is another column called `indicator_code` which symbolizes the category of these debts. Knowing about these various debt indicators will help us to understand the areas in which a country can possibly be indebted to.

In [6]:
SELECT DISTINCT indicator_code AS debt_indicators
FROM international_debt
ORDER BY debt_indicators;

(25 rows affected)

Total execution time: 00:00:00.032

debt_indicators DT.AMT.BLAT.CD DT.AMT.DLXF.CD DT.AMT.DPNG.CD DT.AMT.MLAT.CD DT.AMT.OFFT.CD DT.AMT.PBND.CD DT.AMT.PCBK.CD DT.AMT.PROP.CD DT.AMT.PRVT.CD DT.DIS.BLAT.CD DT.DIS.DLXF.CD DT.DIS.MLAT.CD DT.DIS.OFFT.CD DT.DIS.PCBK.CD DT.DIS.PROP.CD DT.DIS.PRVT.CD DT.INT.BLAT.CD DT.INT.DLXF.CD DT.INT.DPNG.CD DT.INT.MLAT.CD DT.INT.OFFT.CD DT.INT.PBND.CD DT.INT.PCBK.CD DT.INT.PROP.CD DT.INT.PRVT.CD

## 4\. Totaling the Amount of Debt Owed by the Countries

As mentioned earlier, the financial debt of a particular country represents its economic state. But if we were to project this on an overall global scale, how will we approach it?

Let's switch gears from the debt indicators now and find out the total amount of debt (in USD) that is owed by the different countries. This will give us a sense of how the overall economy of the entire world is holding up.

In [8]:
SELECT SUM(debt) as total_debt_USD
FROM international_debt;

(1 row affected)

Total execution time: 00:00:00.018

total_debt_USD 3079734487675.79

## 5\. Country with the Highest Debt

"Human beings cannot comprehend very large or very small numbers. It would be useful for us to acknowledge that fact." - [Daniel Kahneman](https://en.wikipedia.org/wiki/Daniel_Kahneman). That is more than _3 million **million**_ USD, an amount which is really hard for us to fathom.

Now that we have the exact total of the amounts of debt owed by several countries, let's now find out the country that owns the highest amount of debt along with the amount. **Note** that this debt is the sum of different debts owed by a country across several categories. This will help to understand more about the country in terms of its socio-economic scenarios. We can also find out the category in which the country owns its highest debt. But we will leave that for now.

In [11]:
SELECT TOP(1) WITH TIES country_name, ROUND(SUM(debt/1000000000), 2) AS total_debt_Billion_USD
FROM international_debt
GROUP BY country_name
ORDER BY total_debt_Billion_USD DESC;

(1 row affected)

Total execution time: 00:00:00.019

country_name total_debt_Billion_USD China 285.79